File: APP/AlarmSystem/src/main/java/com/chxip/alarmsystem/entity/Response.java
package com.chxip.alarmsystem.entity;

/**
 * 所有接口返回格式
 */
public class Response {

    //错误ID
    private int errorId;

    //具体数据
    private Object data;

    //错误Msg
    private String errorMsg;



    public Response() {
    }
    public Response(int errorId, String errorMsg) {
        this.errorId = errorId;
        this.errorMsg = errorMsg;
    }



    public Response(Object data) {
        this.data = data;
    }

    public int getErrorId() {
        return errorId;
    }

    public void setErrorId(int errorId) {
        this.errorId = errorId;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }
}

File: APP/AlarmSystemAndroid/network/src/main/java/com/chxip/network/entity/Response.kt
package com.chxip.network.entity

/**
 * @ClassName: Response
 * @Description: 接口数据返回基本格式
 * @Author: chxip
 * @CreateDate: 4/20/21 9:59 AM
 */
data class Response<T>(
    //errorCode = 0 代表执行成功，不建议依赖任何非0的 errorCode.
    //errorCode = -1001 代表登录失效，需要重新登录。
    val errorId: Int,

    val errorMsg: String,

    val data: T
)
